src/refmt/eol_detect.ml

```diff
@@ -0,0 +1,19 @@
+type eol =
+  | LF 
+  | CRLF 
+let show eol = match eol with | LF -> "lf" | CRLF -> "crlf"
+let default_eol = match Sys.win32 with | true -> CRLF | _ -> LF
+let crlf_regex = Str.regexp ".*\r\n$"
```

Comment:
Oh, I wonder if we need to be able to do this without `Str` because `Str` cannot be compiled to JavaScript. For example sketch.sh includes the Reason parser as a library then compiles Reason itself into JavaScript. Pretty sure it wouldn't end up pulling in this module but just in case I think we can drop the `Str` dep.

I think after we see the first `\n` when scanning char by char, we can just then check if the char immediately before it was `\r`.

Reply:
Thanks @jordwalke - good catch. I'll implement this w/o using regexp 👍

Reply:
Much simpler now (I think) w/ your sugestion and correctly handles the missed case in the other comment - if there is no newline in the file... we default to platform newline.